src/wristpy/processing/calibration.py

```diff
@@ -0,0 +1,399 @@
+            determined just above the empirically derived  baseline standard deviation
+            due to noise (0.010g)(van Hees et al. 2014). If measuring the noise in a
+            bench-top test, this threshold should be about `1.2 * noise`.
+
+            max_iterations (int): The maximum amount of iterations for the
+            closest_point_fit method. Default is 1000, generally should be left at this
+            value.
+
+            error_tolerance (float): Tolerated level of error, when the
+            closest_point_fit method arrives at this value or better, the process ends.
+            Default is 1e-10. Generally should be left at this value.
+
+            min_calibration_error (float): Minimum acceptable error. If calibration can
+            not reach this threshold it will error.
+
+        Returns:
+            None
+        """
+        self.chunked = chunked
+        self.min_acceleration = min_acceleration
+        self.min_calibration_hours = min_calibration_hours
+        self.min_standard_deviation = min_standard_deviation
+        self.max_iterations = max_iterations
+        self.error_tolerance = error_tolerance
+        self.min_calibration_error = min_calibration_error
+        self.current_iteration = 0
+
+    def run(self, acceleration: models.Measurement) -> models.Measurement:
+        """Runs calibration on acceleration data based on settings.
+
+        If the chunked arguement is set to true, it will run calibration on an initial
+        chunk of the data the size of which is set to min_calibration_hours. If it fails
+        to calibrate based on this subset, it will add 12 hour chunks to the subset
+        until calibration succeeds, or fails on the entire dataset. Otherwise if chunked
+        is false (default) calibration will be conducted on the whole of the dataset.
+        Calibration is successful when scale and offset values, that sufficiently
+        minimize error when applied to the data, are found. The scale and offset values
+        are then applied to every data point in the dataset which is then returned as
+        calibrated models.Measurement object.
+
+        Args:
+            acceleration(Measurement): the accelerometer data containing x,y,z axis
+            data and time stamps.
+
+        Returns:
+            A Measurement object that contains the calibrated acceleration data.
+
+        Raises:
+            ValueError: If the acceleration data does not meet the specified minimum
+            hours of data as given by min_calibration_hours.
+
+            SphereCriteriaError: If the sphere is not sufficiently populated, i.e. every
+            axis does not have at least 1 value both above and below  the + and - value
+            of min_acceleraiton.
+
+            CalibrationError: If the calibration process fails to get below the
+            `min_calibration_error` threshold.
+        """
```

Comment:
Should we add a Reference here (like we are doing with the SleepDetection algorithm?) since there is a published paper on the calibration method?